sympy/polys/domains/field.py

```diff
@@ -68,6 +68,39 @@ def gcd(self, a, b):
 
         return self.convert(p, ring)/q
 
+    def gcdex(self, a, b):
+        """
+        Returns x, y, g such that a * x + b * y == g == gcd(a, b)
+        """
+        try:
+            ring = self.get_ring()
+            found_ring = True
+        except DomainError:
+            found_ring = False
+
+        if (not found_ring) or ring == self:
+            if found_ring:
+                d = self.one
+            else:
+                d = self.gcd(a, b)
+
+            if a == self.zero:
+                if b == self.zero:
+                    return self.zero, self.one, self.zero
+                else:
+                    return self.zero, d/b, d
+            else:
+                return d/a, self.zero, d
+
+        n1, n2 = self.numer(a), self.numer(b)
+        d1, d2 = self.denom(a), self.denom(b)
+        g = ring.gcd(d1, d2)
+        q = d1 * (d2 / g)
+        n1, n2 = n1 * (d2 / g), n2 * (d1 / g)
```

Comment:
We have to avoid using `/` here because we should not assume that it is defined in the ring of integers. We should only use `/` in domain-generic code if we know that the domain is a field which the numerator ring here is not:
https://docs.sympy.org/latest/modules/polys/domainsintro.html#basic-usage-of-domains

If I have ground types gmpy2 then this fails:
```python
In [1]: QQ.gcdex(QQ(3), QQ(5))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 QQ.gcdex(QQ(3), QQ(5))

File ~/current/active/sympy/sympy/polys/domains/field.py:101, in Field.gcdex(self, a, b)
     98 q = d1 * (d2 / g)
     99 n1, n2 = n1 * (d2 / g), n2 * (d1 / g)
--> 101 x, y, d = ring.gcdex(n1, n2)
    102 return self.convert(x, ring)/q, self.convert(y, ring)/q, self.convert(d, ring)/q

File ~/current/active/sympy/sympy/polys/domains/integerring.py:228, in IntegerRing.gcdex(self, a, b)
    226 def gcdex(self, a, b):
    227     """Compute extended GCD of ``a`` and ``b``. """
--> 228     h, s, t = gcdex(a, b)
    229     # XXX: This conditional logic should be handled somewhere else.
    230     if GROUND_TYPES == 'gmpy':

TypeError: gcdext() requires 'mpz','mpz' arguments
```
If I have ground types python then this also fails:
```python
In [1]: QQ.gcdex(QQ(3), QQ(5))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 QQ.gcdex(QQ(3), QQ(5))

File ~/current/active/sympy/sympy/polys/domains/field.py:102, in Field.gcdex(self, a, b)
     99 n1, n2 = n1 * (d2 / g), n2 * (d1 / g)
    101 x, y, d = ring.gcdex(n1, n2)
--> 102 return self.convert(x, ring)/q, self.convert(y, ring)/q, self.convert(d, ring)/q

TypeError: unsupported operand type(s) for /: 'PythonMPQ' and 'float'
```
It only works with ground types flint and that is because in flint `/` always means exact division (exquo) and always preserves the domain: https://github.com/flintlib/python-flint/pull/109

I think that SymPy's domain types should use exact division for `/` everywhere as well but unfortunately with ground types python or gmpy2 there is no way to prevent integer division returning floats. With python-flint and gmpy2:
```python
In [5]: 4 / 2
Out[5]: 2.0

In [6]: mpz(4) / mpz(2)
Out[6]: mpfr('2.0')

In [7]: fmpz(4) / fmpz(2)
Out[7]: 2

In [8]: 1 / 3
Out[8]: 0.3333333333333333

In [9]: mpz(1) / mpz(3)
Out[9]: mpfr('0.33333333333333331')

In [10]: fmpz(1) / fmpz(3)
---------------------------------------------------------------------------
DomainError                               Traceback (most recent call last)
Cell In[10], line 1
----> 1 fmpz(1) / fmpz(3)

File src/flint/types/fmpz.pyx:278, in flint.types.fmpz.fmpz.__truediv__()

DomainError: fmpz division is not exact
```
At least for all domains under SymPy's control we should have `/` mean `exquo`. Given that python-flint already defines it that way I think that the only case we need to be careful about is `ZZ` for when ground types are not flint.

For fraction fields the numerator ring is a polynomial ring and polynomials define `/` but not in the way that we would want here:
```python
In [1]: K = QQ.frac_field(x)

In [2]: K.gcdex(K(x), K(x+1))
Out[2]: (-1, 1, 1)

In [3]: K.numer(K(x))
Out[3]: x

In [4]: K.numer(K(x)) / K.numer(K(x))
Out[4]: 1

In [5]: K.numer(K(x)) / K.numer(K(x+1))
Out[5]: 1

In [7]: K.get_ring().exquo(K.numer(K(x)), K.numer(K(x)))
Out[7]: 1

In [8]: K.get_ring().exquo(K.numer(K(x)), K.numer(K(x+1)))
---------------------------------------------------------------------------
...
ExactQuotientFailed: x + 1 does not divide x in QQ[x]
```
Instead `ring.exquo` should be used. Then again this is what `ring.cofactors` is for (which actually uses `ring.quo` but it should use `ring.exquo`).

I'm not sure if there is one already but we need a test somewhere that comprehensively covers all domain types and checks their behaviour for divisibility with `div` `quo`, `rem`, `exquo`, `gcd`, `gcdex`, `/`, `//`, `%` and `divmod`.